### PR TITLE
#1845: Unexpected version comparison result when comparing versions with stars

### LINKED ISSUE #1845: Unexpected version comparison result when comparing versions with stars
### Actual behavior

IDEasy supports comparing two versions which returns which version of the two is greater.

For example, when comapring version `1.0` with version `1.1`, IDEasy will determine, that version `1.0` is `LESS` than version `1.1`. This generally works well. However, issues occur, when comparing version ranges that include stars.

For example, when comparing version `1.*`  with version `1.1`, IDEasy will report that version `1.*` is `LESS` than version `1.1`. This is unexpected, since we use the `*` to indicate the latest version.

### Reproduce

This cannot be replicated from the command line as this functionality is not exposed to the user.

### Expected behavior

IDEasy should report that versions like `x.*` are greater than version `x.y`.

### IDEasy status

```bash
IDE_ROOT is set to C:\Users\areinick\projects
IDE_HOME is set to C:\Users\areinick\projects\IDEasy
Your version of IDEasy is 2026.04.002.
Your are using the latest version of IDEasy and no update is available.
Your operating system is windows(10.0)@x64 [Windows 11@amd64]
You are online.
Found bash executable at: C:\Program Files\Git\usr\bin\bash.exe
Found git executable at: C:\Program Files\Git\mingw64\bin\git.exe
Your settings are up-to-date.
Successfully completed ide (status)
```

### Related/Dependent issues

_No response_

### Comments/Hints

_No response_

### PR DESCRIPTION
**Pull request by areinicke**
### Implemented changes:

* Implemented several changes to make version comparisons more robust, exceeding the issues originally described in #1845 :

1. Comparisons of versions `x.*` and `x.y` should return a `GREATER_UNSAFE` result
2. Comparisons of versions `x.*` and `x.*!` should now return a `LESS_UNSAFE` result
3. Several checks were introduced to make this work with both number and letter based version names

---

## Checklist for this PR

Make sure everything is checked before merging this PR. For further info please also see
our [DoD](https://github.com/devonfw/IDEasy/blob/main/documentation/contributing/DoD.adoc).

- [x] When running `mvn clean test` locally all tests pass and build is successful
- [x] PR title is of the form `#«issue-id»: «brief summary»` (e.g. `#921: fixed setup.bat`). If no issue ID exists, title only.
- [x] PR top-level comment summarizes what has been done and contains link to addressed issue(s)
- [x] PR and issue(s) have suitable labels
- [x] Issue is set to `In Progress` and assigned to you *or* there is no issue (might happen for very small PRs)
- [x] You followed all [coding conventions](https://github.com/devonfw/IDEasy/blob/main/documentation/coding-conventions.adoc)
- [x] You have added the issue implemented by your PR in [CHANGELOG.adoc](https://github.com/devonfw/IDEasy/blob/main/CHANGELOG.adoc) unless issue is labeled
  with `internal`

### CODE REVIEW

File: cli/src/main/java/com/devonfw/tools/ide/version/VersionSegment.java [hunk 1]
```diff
@@ -221,6 +221,14 @@ public VersionComparisonResult compareVersion(VersionSegment other) {
     if (other == null) {
       return VersionComparisonResult.GREATER_UNSAFE;
     }
+
+    // Check if one version segment is * or *! while the other has no pattern. If so, no further comparison is needed
+    if (this.letters.isEmpty() && isPattern() && !other.letters.isEmpty()) {
+      return VersionComparisonResult.GREATER_UNSAFE;
+    } else if (other.letters.isEmpty() && other.isPattern() && !this.letters.isEmpty()) {
+      return VersionComparisonResult.LESS_UNSAFE;
+    }
```
Comment (hohwille) [inline cli/src/main/java/com/devonfw/tools/ide/version/VersionSegment.java:230]: I am slightly wondering how this is related to the `letters` of the `VersionSegment`...
What if I compare `1.0alp*` with `1.0alpha`?
Or `1.0alpha*` and `1.0alpha1`?
Or `1.0*` and `1.0`?
Your added code is surely not wrong and I am happy to merge.
My comment is just to say that we might still have some edge-cases not properly covered.

**Review by hohwille — APPROVED** [review]
@areinicke thank you very much for noticing this problem and creating this PR with test + fix. 👍 
I have added some comment. But I will resolve it and simply merge.
However, please have a look and if my thoughts are correct, we can later add another PR with further improvements...
